### PR TITLE
Test refactor 2

### PR DESCRIPTION
**Pull request by bamandel**
Cleaned up tests and changed from a single large mock item to 4 individual ones. Separate files for Entity, GlobalEntity, RestModel, and GlobalRestModel.

### CODE REVIEW

File: src/test/java/com/blackducksoftware/integration/hub/alert/mock/entity/MockHipChatEntity.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    private MockHipChatEntity(final Integer roomId, final Boolean notify, final String color, final Long id) {
+        this.roomId = roomId;
+        this.notify = notify;
+        this.color = color;
+        this.id = id;
+    }
+
+    @Test
+    public void tes() throws JSONException {
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/mock/entity/MockHipChatEntity.java]: Can you give this test a real name?

File: src/test/java/com/blackducksoftware/integration/hub/alert/mock/entity/global/MockGlobalHubEntity.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    private MockGlobalHubEntity(final Integer hubTimeout, final String hubUsername, final String hubPassword, final Long id) {
+        super();
+        this.hubTimeout = hubTimeout;
+        this.hubUsername = hubUsername;
+        this.hubPassword = hubPassword;
+        this.id = id;
+    }
+
+    @Test
+    public void test() throws JSONException {
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/mock/entity/global/MockGlobalHubEntity.java]: Can you give this test a real name?

File: src/test/java/com/blackducksoftware/integration/hub/alert/mock/entity/MockHipChatEntity.java (hunk 1 above)
Comment (bamandel) [reply src/test/java/com/blackducksoftware/integration/hub/alert/mock/entity/MockHipChatEntity.java]: These tests will be gone once I change from interface to an abstract class.

**Comment by bamandel** [comment]
The main things that I've added are tests for processors and just the global controllers at the moment. Everything else is really a refactor of the old testing code I wrote.

File: src/main/java/com/blackducksoftware/integration/hub/alert/web/model/distribution/CommonDistributionConfigRestModel.java [hunk 3]
```diff
@@ -24,10 +24,9 @@
 
 import java.util.List;
 
-import org.apache.commons.lang3.builder.RecursiveToStringStyle;
-import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
-
 import com.blackducksoftware.integration.hub.alert.web.model.ConfigRestModel;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/hub/alert/web/model/distribution/CommonDistributionConfigRestModel.java]: Is there a son equivalent to this?

File: src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyOverrideProcessorTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import com.blackducksoftware.integration.hub.model.view.ComponentVersionView;
+import com.blackducksoftware.integration.hub.model.view.PolicyRuleView;
+import com.blackducksoftware.integration.hub.notification.processor.MapProcessorCache;
+import com.blackducksoftware.integration.test.TestLogger;
+
+public class PolicyOverrideProcessorTest {
+
+    @Test
+    public void testProcessRemoveEvent() throws HubIntegrationException, URISyntaxException {
+        // Test will break if ListProcessorCache is used because NotificationEvents does not have an equals method defined
```
Comment (gkillough) [inline src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyOverrideProcessorTest.java]: Why not add an equals method?

File: src/main/java/com/blackducksoftware/integration/hub/alert/web/model/distribution/CommonDistributionConfigRestModel.java (hunk 3 above)
Comment (bamandel) [reply src/main/java/com/blackducksoftware/integration/hub/alert/web/model/distribution/CommonDistributionConfigRestModel.java]: I wasn't able to find a good replacement. If I were to actually use Gson on the object in its toString method, I would also need to change the expected json output for all our tests. If there are empty or null values in an object that has Gson ran on it, then the values aren't added to the new json string where our tests expect value: null

File: src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyOverrideProcessorTest.java (hunk 4 above)
Comment (bamandel) [reply src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyOverrideProcessorTest.java]: NotificationEvent is an object from another library. So we'd have to go through an entire release just to fix a test. We use the MapCache in alert so we should be using that in the test anyway, but maybe in the future we may want to add an equals method

Comment (gkillough) [reply src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyOverrideProcessorTest.java]: So then why is the comment necessary if we use a MapCache anyway?

Comment (bamandel) [reply src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyOverrideProcessorTest.java]: I wanted to worn people about a possible issue if it were changed. I think we should keep the comment, but maybe we can move it elsewhere

File: src/main/java/com/blackducksoftware/integration/hub/alert/web/model/distribution/CommonDistributionConfigRestModel.java (hunk 3 above)
Comment (psantos1113) [reply src/main/java/com/blackducksoftware/integration/hub/alert/web/model/distribution/CommonDistributionConfigRestModel.java]: I don't want to introduce a different JSON serialization technology just to support tests.  I want to be consistent I would say go back to GSON. The tests for empty fields is handled by the entity tests where the fields are set to null.  The method: 
getEmptyEntityJson method in the test code should be in a base class and return the json string of "{}" which is an empty JSON object.

File: src/test/java/com/blackducksoftware/integration/hub/alert/mock/entity/MockHipChatEntity.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        final JsonObject json = new JsonObject();
+        json.addProperty("roomId", roomId);
+        json.addProperty("notify", notify);
+        json.addProperty("color", color);
+        json.addProperty("id", id);
+        return json.toString();
+    }
+
+    @Override
+    public String getEmptyEntityJson() {
```
Comment (psantos1113) [inline src/test/java/com/blackducksoftware/integration/hub/alert/mock/entity/MockHipChatEntity.java]: These methods should be removed in classes like this.  a base class can return "{}" and empty json array for testing.

**Comment by bamandel** [comment]
Added a couple more tests and resolved some issues with toString methods. **I made sensitive fields transient**. I did this because it worked best with Gson and also allows to possibly remove the need for sensitive field functionality. I verified that the slack integration still worked after this change (I would test hipchat which has transient fields but it's broken at the moment)

**Pull request merged**